### PR TITLE
Added in very specific support for from_json to a Map<String,String>

### PR DESCRIPTION
**Pull request by revans2**
This depends on https://github.com/rapidsai/cudf/pull/11431

### CODE REVIEW

File: docs/configs.md [hunk 1]
```diff
@@ -243,6 +243,7 @@ Name | SQL Function(s) | Description | Default Value | Notes
 <a name="sql.expression.IsNaN"></a>spark.rapids.sql.expression.IsNaN|`isnan`|Checks if a value is NaN|true|None|
 <a name="sql.expression.IsNotNull"></a>spark.rapids.sql.expression.IsNotNull|`isnotnull`|Checks if a value is not null|true|None|
 <a name="sql.expression.IsNull"></a>spark.rapids.sql.expression.IsNull|`isnull`|Checks if a value is null|true|None|
+<a name="sql.expression.JsonToStructs"></a>spark.rapids.sql.expression.JsonToStructs|`from_json`|Returns a struct value with the given `jsonStr` and `schema`|true|None|
```
Comment (jlowe) [inline docs/configs.md:246]: The code says this is disabled by default, but the docs show it is enabled by default?

Comment (revans2) [reply docs/configs.md:246]: Good catch. I need to rerun the doc generation.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala [hunk 2]
```diff
@@ -3476,6 +3476,20 @@ object GpuOverrides extends Logging {
...
+        TypeSig.MAP.nested(TypeSig.STRING),
+        (TypeSig.STRUCT + TypeSig.MAP + TypeSig.ARRAY).nested(TypeSig.all),
+        Seq(ParamCheck("jsonStr", TypeSig.STRING, TypeSig.STRING))),
+      (a, conf, p, r) => new UnaryExprMeta[JsonToStructs](a, conf, p, r) {
+        override def tagExprForGpu(): Unit =
+          GpuJsonScan.tagJsonToStructsSupport(a.options, this)
+
+        override def convertToGpu(child: Expression): GpuExpression =
+          GpuJsonToStructs(a.schema, a.options, child, a.timeZoneId)
+      }).disabledByDefault("Parsing JSON from a column has a large number of issues and " +
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala:3491]:
```suggestion
      }).disabledByDefault("parsing JSON from a column has a large number of issues and " +
```

**Review by jlowe — APPROVED** [review]
Small capitalization nit, lgtm.

**Review by jlowe — APPROVED** [review]

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@jlowe please take another look. I had to upmerge to fix a small issue with auto-generated documentation that didn't auto-merge correctly.

**Review by jlowe — APPROVED** [review]

**Comment by sameerz** [comment]
build

**Pull request merged**